spec/system/index_spec.rb

```diff
@@ -87,19 +84,61 @@
   let(:user) { create :user }
   let!(:route) { create :route }
   let!(:bus_stop) { create :bus_stop }
-  let!(:bus_stops_route) { create :bus_stops_route, bus_stop: bus_stop, route: route }
+  let!(:bus_stops_route) do
+    create :bus_stops_route,
+           bus_stop: bus_stop,
+           route: route
+  end
+  let(:incorrect_route_number) { '-1' }
   before :each do
     when_current_user_is user
     visit root_url
   end
   context 'route from the dropdown' do
-    it 'redirects to bus stops by status' do
+    before :each do
       within 'form', text: 'Select a route' do
         select route.number, from: 'Select a route'
         click_button 'View stops'
       end
+    end
+    it 'redirects to bus stops by status' do
       expect(page).to have_content route.number
       expect(page.current_path).to end_with by_status_bus_stops_path
     end
+    context 'click view by route order' do
+      before :each do
+        click_link 'View by route order'
+      end
+      it 'redirects to bus stops by sequence' do
+        expect(page).to have_content route.number
+        expect(page.current_path).to end_with by_sequence_bus_stops_path
+      end
+    end
+  end
+  context 'incorrect route in params' do
```

Comment:
I would rename this as system tests are all about user interaction and users can't directly change params.

Reply:
Good question, actually, as we've established that this isn't something that can really happen as a result of user interaction.

I would argue that a controller test might be more appropriate here, and in general when testing any inputs that a controller can receive but which the corresponding view cannot produce.

Reply:
Alright, will do.